Make CI Great Again

ozymandias-rex opened the pull request:
This is a PR to clean up code base (i.e. lints , fmt , etc). Basically Make CI Great Again

BatiGencho committed:
added integration test suite and moved integration tests into it

scottmillner committed:
feature: fetch and store block tx ids in memory

ozymandias-rex committed:
chore: fmt lints

ozymandias-rex committed:
chore: fmt lints

ozymandias-rex committed:
chore: clippy lints consensus

ozymandias-rex committed:
chore: clippy lints consensus

ozymandias-rex committed:
fmt

ozymandias-rex committed:
fmt

BatiGencho committed:
added jwt authentication to btc server and client

ozymandias-rex committed:
lint: rpc

scottmillner committed:
feature: make get_btc_fee_rate dynamic

ozymandias-rex committed:
chore: more lint fixes

ozymandias-rex committed:
ci: bump

ozymandias-rex committed:
chore: run ci on pushes for testing

ozymandias-rex committed:
chore: fmt

BatiGencho committed:
frost signing reworked

arminsabouri commented on .github/workflows/lint.yml:
```diff
@@ -108,7 +108,7 @@ jobs:
         with:
           components: rustfmt
-      - run: cargo fmt --all --check
+      - run: cargo +nightly fmt --all --check
```
Do we need nightly? I think stable should be fine

BatiGencho replied:
Of the same opintion. @ozymandias-rex if we merge this one in it would break all our MRs due to linting changes and formatting.

ozymandias-rex replied:
The reason I did this , was to be consistent with the [Makefile's fmt](https://github.com/botanix-labs/Macbeth/blob/feature/poA-consensus/Makefile#L311). @BatiGencho had mentioned that we should run this before commits , and it contradicts what is in the ci. Should we change the Makefile's fmt to not use nightly ?

BatiGencho replied:
@ozymandias-rex I am ok with nightly  but suggest we merge the remaining 2 mrs, then you take this MR, run fmt on the code again and we merge it then

ozymandias-rex replied:
In addition to this , most of the lints defined in the[ rustfmt.toml](https://github.com/botanix-labs/Macbeth/blob/feature/poA-consensus/rustfmt.toml) file are only available in nightly 

```
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `binop_separator = Back`, unstable features are only available in nightly channel.
Warning: can't set `trailing_semicolon = false`, unstable features are only available in nightly channel.
Warning: can't set `trailing_comma = Vertical`, unstable features are only available in nightly channel.
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `doc_comment_code_block_width = 100`, unstable features are only available in nightly channel.
Warning: can't set `comment_width = 100`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `binop_separator = Back`, unstable features are only available in nightly channel.
Warning: can't set `trailing_semicolon = false`, unstable features are only available in nightly channel.
Warning: can't set `trailing_comma = Vertical`, unstable features are only available in nightly channel.
Warning: can't set `wrap_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `format_code_in_doc_comments = true`, unstable features are only available in nightly channel.
Warning: can't set `doc_comment_code_block_width = 100`, unstable features are only available in nightly channel.
Warning: can't set `comment_width = 100`, unstable features are only available in nightly channel.
Warning: can't set `imports_granularity = Crate`, unstable features are only available in nightly channel.
Warning: can't set `binop_separator = Back`, unstable features are only available in nightly channel.
Warning: can't set `trailing_semicolon = false`, unstable features are only available in nightly channel.
Warning: can't set `trailing_comma = Vertical`, unstable features are only available in nightly channel.
```

ozymandias-rex replied:
> @ozymandias-rex I am ok with nightly but suggest we merge the remaining 2 mrs, then you take this MR, run fmt on the code again and we merge it then

ack

arminsabouri replied:
ack

arminsabouri reviewed (commented):
Alot of these formatting changes seem inconsistent with upstream reth team. Is it possible we are using different reth fmt formatting rules? Can we commit formatting rules?

ozymandias-rex committed:
chore: dont run lint on all features and examples

ozymandias-rex committed:
chore: dont run lint on all features and examples

ozymandias-rex committed:
chore: use stable fmt

ozymandias-rex committed:
chore: use stable fmt

scottmillner committed:
validate psbt fee rate within range

BatiGencho committed:
added dkg and signing integration tests

scottmillner committed:
refactor: process pegouts before building block header

BatiGencho committed:
added minting contract functionality

arminsabouri committed:
update makefile to use regtest setting

arminsabouri committed:
refactor(rpc): default network should be regtest

arminsabouri committed:
test(frost_integration): generate gateway address and send functions

arminsabouri committed:
chore: contract abi location should be relative

arminsabouri committed:
feature: add serialize() to pegin meta

arminsabouri committed:
test(frost_integration): WIP - fill in pegin meta structure

BatiGencho committed:
added remaining test flow

arminsabouri committed:
test(frost_integration): pull down conf header and generate pmt

scottmillner committed:
chore: update default pegin network to regtest

Co-authored-by: Armin Sabouri <armins88@gmail.com>

scottmillner committed:
fix(poa): storage deadlock in epoch_manager

Co-authored-by: Armin Sabouri <armins88@gmail.com>

scottmillner committed:
fix(poa): witness data should never be a empty vec only None

Co-authored-by: Armin Sabouri <armins88@gmail.com>

scottmillner committed:
fix(poa): if there are no pegouts do not call `send_pegout()`

Co-authored-by: Armin Sabouri <armins88@gmail.com>

scottmillner committed:
fix: refactor epoch_pegouts

Co-authored-by: Armin Sabouri <armins88@gmail.com>

arminsabouri committed:
test(frost_integration): fix various problems with pegin call to mint contract

arminsabouri committed:
test(frost_integration): fix various problems with pegin call to mint contract

scottmillner committed:
feature(poa): add signer finalize method to `extended_client`

Co-authored-by: Armin Sabouri <armins88@gmail.com>

scottmillner committed:
feature(poa): WIP- allow for the coordinator to be a signer
this commit is a work in progress that comments out many of the state transition checks

Co-authored-by: Armin Sabouri <armins88@gmail.com>

scottmillner committed:
chore(btc_server): log the requests coming to the coordinator

Co-authored-by: Armin Sabouri <armins88@gmail.com>

arminsabouri committed:
chore: remove duplicate defs for start-poa commands

arminsabouri committed:
feature(btc_server): broadcast finalized tx before removing old utxos

ozymandias-rex committed:
feat: botanix tests paying; todo : fix broken reth tests

ozymandias-rex committed:
fix: merge changes

ozymandias-rex committed:
chore: fmt

arminsabouri committed:
chore(poa): remove unused `finalize_signer()` function

arminsabouri committed:
fix(edh): update `HAS_WITNESS_DATA` optional fields bitmask

arminsabouri committed:
fix(poa): include witness data during block creation

arminsabouri committed:
feature(poa): move get epoch pegouts to a util method

commit is also responsible for calling btc_server `signer_finalize()`
after retrieving a epoch block

ozymandias-rex committed:
feat: final rebases and running tests

arminsabouri commented on Makefile:
```diff
@@ -398,7 +399,7 @@ start-test-suite:
 start-btc-server-1:
 	cd ./bin/btc-server && \
-	cargo run --bin btc-server -- --network testnet --identifier 0 --address 0.0.0.0:8080 --db "./db1" --min-signers 2 --max-signers 2 --toml ./config.toml  --fee-rate-diff-percentage 30 --bitcoind-url localhost:18443 --bitcoind-user foo --bitcoind-pass bar --jwt-secret "${NODE_1_DIR}/jwt.hex" --fall-back-fee-rate-sat-per-vbyte 5
+	cargo run --bin btc-server -- --network testnet --identifier 0 --address 0.0.0.0:8080 --db "./db1" --min-signers 1 --max-signers 1 --toml ./config.toml  --fee-rate-diff-percentage 30 --bitcoind-url localhost:18443 --bitcoind-user foo --bitcoind-pass bar --jwt-secret "${NODE_1_DIR}/jwt.hex" --fall-back-fee-rate-sat-per-vbyte 5 
```
im not sure if min signers and max signers both being 1 will be accepted by the frost library. We should have a check for this in btc_server if we dont already

ozymandias-rex replied:
found this out the hard way . it has to be 2

arminsabouri commented on Makefile:
```diff
@@ -22,7 +22,8 @@ endif
 # Cargo profile for builds. Default is for local builds, CI uses an override.
 PROFILE ?= release
-
+NODE_1_DIR="bin/btc-server/"
+NODE_1_DIR="bin/btc-server/"
```
defined twice

arminsabouri commented on bin/btc-server/bin/btc-server/jwt.hex:
```diff
@@ -0,0 +1 @@
+fc81b3bc2a81f7d84d5a8ec75b3e75fbdd014b35b29c84206c36d8d4ce301c44
```
Should we commit these? I ussually just set the env var before running the make file

ozymandias-rex replied:
My reasoning was that it's a dummy var we never use, so harmless. There is the template file which we commit directly and could use.

arminsabouri commented on bin/btc-server/src/util.rs:
```diff
@@ -279,6 +279,24 @@ pub fn add_partial_signature_to_psbt(
     }
 }
 
+pub(crate) fn fee_rate_into_u32(fee_rate: u64) -> u32 {
+    match <u64 as TryInto<u32>>::try_into(fee_rate) {
+        Ok(fee_rate) => fee_rate,
+        Err(e) => {
+            error!("Failed to convert fee rate to u32: {}", e);
+            0
+        }
+    }
+}
+
+pub(crate) fn sats_kb_to_vb_u32(fee_rate: u32) -> u32 {
+    fee_rate / 1000
+}
+
+pub(crate) fn sats_kb_to_vb_u64(fee_rate: u64) -> u64 {
+    fee_rate / 1000
+}
```
I don't think this is being used anywhere. Safe to delete.

arminsabouri reviewed (commented):
Just some initial feedback

ozymandias-rex committed:
feat: patch ci , use nightly
